texar/modules/connectors/connectors.py

```diff
@@ -0,0 +1,786 @@
+            # sample == Tensor([batch_size_, 10])
+
+    Args:
+        output_size: Size of output **excluding** the batch dimension. For
+            example, set ``output_size`` to ``dim`` to generate output of
+            shape ``[batch_size, dim]``.
+            Can be an ``int``, a tuple of ``int``, a ``torch.Size``, or
+            a ``tuple`` of ``torch.Size``.
+            For example, to transform inputs to have decoder state size, set
+            :python:`output_size=decoder.state_size`.
+        hparams (dict, optional): Hyperparameters. Missing
+            hyperparamerter will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter sturcture and
+            default values.
+        distribution: A instance of subclass of
+            :torch:`distributions.distribution.Distribution`,
+            Can be a class, its name or module path, or a class instance.
+        distribution_kwargs (dict, optional): Keyword arguments for the
+            distribution constructor. Ignored if :attr:`distribution` is a
+            class instance.
+    """
+
+    def __init__(self,
+                 output_size: OutputSize,
+                 hparams: HParamsType = None,
+                 distribution: Union[Type[T], T, str] = 'MultivariateNormal',
+                 distribution_kwargs: Optional[Dict[str, Any]] = None,):
+        ConnectorBase.__init__(self, output_size, hparams)
+        if distribution_kwargs is None:
+            distribution_kwargs = {}
+        self._dstr = utils.check_or_get_instance(
+            distribution, distribution_kwargs,
+            ["torch.distributions", "torch.distributions.multivariate_normal",
+             "texar.custom"])
+
+    @staticmethod
+    def default_hparams() -> dict:
+        r"""Returns a dictionary of hyperparameters with default values.
+
+        .. code-block:: python
+
+            {
+                "activation_fn": "texar.core.layers.identity",
+                "name": "reparameterized_stochastic_connector"
+            }
+
+        Here:
+
+        "activation_fn" : str
+            The activation function applied to the outputs of the MLP
+            transformation layer. Can be a function,
+            or its name or module path.
+        "name" : str
+            Name of the connector.
+        """
+        return {
+            "activation_fn": "texar.core.layers.identity",
+            "name": "reparameterized_stochastic_connector"
+        }
+
+    def forward(self, # type: ignore
+                transform: bool = True,
+                num_samples: Optional[Union[int, torch.Tensor]] = None
+                ) -> Tuple[Any, Any]:
+        r"""Samples from a distribution and optionally performs transformation
+        with an MLP layer.
+        The distribution must be reparameterizable, i.e.,
+        :python:`distribution.reparameterization_type = FULLY_REPARAMETERIZED`.
```

Comment:
PyTorch distributions do not have such attributes. In PyTorch, whether a distribution is reparameterizable can be inferred by accessing an undocumented class attribute: `Distribution.has_rsample`.

Reply:
Thanks! That is not a easy-to-find one.